.github/workflows/build.yaml

```diff
@@ -7,10 +7,65 @@ on:
   pull_request:
     branches:
       - master
-
+env:
+  cache-path: |
+    .git/modules
+    binutils/.git
+    gdb/.git
+    gcc/.git
+    llvm/.git
+    newlib/.git
+    glibc/.git
+    musl/.git
+    uclib-ng/.git
+    dejagnu/.git
+    pk/.git
+    qemu/.git
+    spike/.git
 jobs:
+  cache:
+    name: Update Submodule Cache
+    runs-on: ubuntu-24.04
+    outputs:
+      key: submodules-${{ steps.submodule-hash.outputs.HASH }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Generate Submodule Hash
+        id: submodule-hash
+        run: echo "HASH=$(git submodule | sha1sum | head -c 40)" >> $GITHUB_OUTPUT
+
+      - name: Check is Cache Exists for Exact Submodule Configuration
```

Comment:
Typo? "is" should be "if"?

Very minor cosmetic (pedantic?!) issue - why the capitalisation of words in comments (here and elsewhere)?

Reply:
That's a correct typo. I was using title case because they appear in the CI as the titles of each step. But if the standard here is all lower case or sentence case I'll happily switch to that.

Reply:
> Thanks for catching the typo. I was using title case because they appear in the CI as the titles of each step. But if the standard here is all lower case or sentence case I'll happily switch to that.

Ah - they're names of the steps/tasks. I mistakenly thought that they were comments. If that's the convention then it makes sense to adhere to it. But should every word start with a capital letter? Not all do - e.g. "Check if Cache Exists for Exact Submodule Configuration", "If no Cache Hit, Update Cache", "Clone submodules", etc. Maybe prepositions are exempt from the rule? :-)